### PR TITLE
#171380739 setup sequelize

### PR DESCRIPTION
**Pull request by since9629**
#### What does this PR do?
Setup sequelize as a an ORM to the project
#### Description of Task to be completed?
- create a .sequelizerc file to hold configurations that sequelize-cli needs
- Install `sequelize` and its dependencies
- Install `sequelize-cli` for working with sequelize using command line
- Initialize sequelize with command `npx sequelize-cli init`
- Create a database with command `npx sequelize-cli db:create`
- Generate a dummy user model and migrations with command `npx sequelize-cli model:generate --name User --attributes email:string,password:string`
- Run user migration with command `npx sequelize-cli db:migrate`
#### How should this be manually tested?
- After cloning the project repo, run `npm i` inside the cloned repo folder
- Make sure to rename .env.example to .env, 
- In .env file, provide all values for variables under `#Development database`  and `#General`
- Finally, run the following commands respectively:
`git fetch`
`git checkout ch-setup-sequelize-171380733`
`npx sequelize-cli db:create` to create a database
`npx sequelize-cli db:migrate` to create a user table
- If the above steps have been followed correctly, you should see a new user table created in your database
#### Any background context you want to provide?
Refer to [the sequelize-cli](https://github.com/sequelize/cli/tree/master/docs) and [sequelize](https://sequelize.org/v5/) docs for more.
#### What are the relevant pivotal tracker stories?
#171380739
#### Screenshots (if appropriate)
N/A

### CODE REVIEW

**Review by muchristian — APPROVED** [review]

File: src/database/models/index.js [hunk 1]
```diff
@@ -1,30 +1,39 @@
...
-const pool = new pg.Pool({
-  connectionString: DATABASE_URI,
-});
+require('dotenv').config();
 
-pool.on('connect', () => {
-});
+const basename = _basename(__filename);
+const env = process.env.NODE_ENV || 'development';
+const config = require('../config.js')[env];
```
Comment (ricahcyuzuzo) [inline src/database/models/index.js:9]: Please use ES6

Comment (since9629) [reply src/database/models/index.js:9]: @muchristian at what line are you referring to exactly?

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Well done, this is the great work you have done, I appreciate it sincerely, but I have few suggestions for you, maybe you will explain more to clarify, or you can change something to make it works.

* A. On the file `src/database/config.js` line 4 - 11, you declared an object called `development` and you gave it those properties which seem to help in `db creation` , `db connection`, and `migrations`. For me it didn't work but when I moved those properties out of that object, it worked, so I think you can help me to understand why did you put those properties in the object, if it works for you maybe it is not compatible with my system, I use `Linux mint 19.1 Tessa`. As you see on the screenshot
Screenshot from 2020-03-04 08-42-51
 that is the error I am getting when I run these commands 
```javascript
./node_modules/.bin/sequelize-cli db:create to create a database
./node_modules/.bin/sequelize-cli db:migrate to create a user table
```
, before editing that file.

* But after editing that file `(src/database/config.js)` like this : 
```javascript
module.exports = {
  username: process.env.DEV_DB_USERNAME,
  password: process.env.DEV_DB_PASSWORD,
  database: process.env.DEV_DB_NAME,
  host: process.env.DEV_DB_HOSTNAME,
  port: process.env.DEV_DB_PORT,
  dialect: 'postgres'
};
```
=>It gave me this output : 
Screenshot from 2020-03-04 08-50-01
which seems to me that it worked because I can see now table users is created.

B. On the PR in the *How should this be manually tested?* clause, I suggest that you add an explanation of this line `database: process.env.DEV_DB_NAME`. I mean the DEV_DB_NAME, because someone can put there an already created database, and tend to run this command `./node_modules/.bin/sequelize-cli db:create` which is supposed to create the database as it is explained in this file `./node_modules/.bin/sequelize-cli` line 76 (see it on the screenshot below)
Screenshot from 2020-03-04 08-58-20
=>So what I suggest here, you need to clarify that, if the `db` exists I think there is no need to run that command of creating it, but if  there is not already created `db` then run it, or if you want to create a `dev db` then on this line `database: process.env.DEV_DB_NAME` put the desired `dev db_name`

**Review by ricahcyuzuzo — APPROVED** [review]

**Review by descholar-ceo — APPROVED** [review]

**Review by Muhire-Josue — APPROVED** [review]
LGTM, great job @karamuka

File: README.md [hunk 2]
```diff
@@ -30,4 +30,70 @@ _If the aforementioned link says *ACCESS DENIED*, then contact me _[here](mailto
...
+
+The .sequelizerc file will in turn be used by sequelize-cli to setup the the sequelize folder structure.
+
+### 1. Configure environment variables
+
+The following are example environment variables needed:
+
+#### Database
+
+* DEV_DB_USERNAME=postgres
```
Comment (mifeille) [inline README.md]: you don't have to add values to those variables cause, for example, their username might be different from what you wrote.

**Review by mifeille — APPROVED** [review]

**Pull request merged**